src/Avalonia.Controls/Primitives/RangeBase.cs

```diff
@@ -167,13 +176,9 @@ protected override void OnInitialized()
         /// Throws an exception if the double value is NaN or Inf.
         /// </summary>
         /// <param name="value">The value.</param>
-        /// <param name="property">The name of the property being set.</param>
-        private static void ValidateDouble(double value, string property)
+        private static bool ValidateDouble(double value)
         {
-            if (double.IsInfinity(value) || double.IsNaN(value))
-            {
-                throw new ArgumentException($"{value} is not a valid value for {property}.");
-            }
+            return (!double.IsInfinity(value) || !double.IsNaN(value));
```

Comment:
IMO it'd be more readable to have
```cs
return (!double.IsInfinity(value) && !double.IsNaN(value));
```
then have
```cs
if (ValidateDouble(value))
{
	value = _field;
}
```

Also, don't forget to update the XML comments

Reply:
It's semantic because `ValidateDouble` to me indicates that it returns true if it is a "valid" double value. But yeah, it'll be better if i rename the method instead and do it the other way